Debian #157792, #160578, #184218; handle PAM's AUTHINFO_UNAVAIL in incorrect
password error handler.

Index: gdm-2.18.1/daemon/verify-pam.c
===================================================================
--- gdm-2.18.1.orig/daemon/verify-pam.c	2007-05-02 18:29:52.000000000 +0200
+++ gdm-2.18.1/daemon/verify-pam.c	2007-05-02 18:34:14.000000000 +0200
@@ -1166,8 +1166,10 @@
     if ( ! error_msg_given &&
 	gdm_slave_action_pending ()) {
 	    /* I'm not sure yet if I should display this message for any other issues - heeten */
+	    /* at least AUTHINFO_UNAVAIL -- it's what user unknown is! */
 	    if (pamerr == PAM_AUTH_ERR ||
-		pamerr == PAM_USER_UNKNOWN) {
+		pamerr == PAM_USER_UNKNOWN ||
+		pamerr == PAM_AUTHINFO_UNAVAIL) {
 		    gboolean is_capslock = FALSE;
 		    const char *basemsg;
 		    char *msg;
